tensorboard/backend/event_processing/event_accumulator.py

```diff
@@ -366,7 +365,7 @@ def _ProcessEvent(self, event):
             )
             if self._source_writer and self._source_writer != new_source_writer:
                 # This should not happen.
```

Comment:
I would also remove this comment, as it could theoretically happen in the situation Nick described in #6014.